### PR TITLE
Updating README & CONTRIBUTING guidelines and adding an initial CHANGELOG

### PR DESCRIPTION
**Pull request by sts**
- Update README
- Sort out sections to CONTRIBUTING & to a new CHANGELOG

- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+* Update libinjection-go [PR #157](https://github.com/corazawaf/coraza/pull/157)
+* Better debug logging
+* New error logging (like modsecurity)
+
+NOTES:
+
+**Migrate from Corza v1**
+  * Rollback SecAuditLog to the legacy syntax (serial/concurrent)
+  * Attach an error log handler using ```waf.SetErrorLogCb(cb)``` (optional)
+  * the function Transaction.Clean() must be used to clear transaction data, files and take them back to the sync pool.
```
Comment (fzipi) [inline CHANGELOG.md:28]: Maybe `quoting` the name of the function?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  * If you are using low level APIs check the complete changelog as most of them were removed.
+  * OWASP CRS does not require any external dependency anymore
```
Comment (fzipi) [inline CHANGELOG.md:30]: * OWASP CRS does not require external dependencies anymore

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,37 +1,82 @@
...
-* Ensure the PR description clearly describes the problem and solution. Include the relevant issue number if applicable.
+* Security related issues are covered by the [Security Policy](https://github.com/corazawaf/coraza/blob/v2/master/SECURITY.md)
+* Make sure you test against the latest version, it's possible the issue was
+  already fixed. However if you are on an older version of Coraza and feel the
+  issue is critical, please let us know.
+* Check existing [Issues](https://github.com/corazawaf/coraza/issues) (open and closed) to ensure it was not already reported.
+* Provide a detailed description and a reproducible test case in a new [Issue](https://github.com/corazawaf/coraza/issues/new).
+  Be sure to include as much relevant information as possible, a **code sample** or an **test case** demonstrating the fault helps us to reproduce your problem.
 
-* Before submitting, please commit using [pre-commit](https://pre-commit.com/) to validate your PR passes the quality tests
```
Comment (fzipi) [inline CONTRIBUTING.md:18]: We use the [pre-commit](https://pre-commit.com/) framework to perform quality tests before sending them for review. It is recommended you install it and use it before sending any patches.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,37 +1,82 @@
...
+## Patches
+Did you write a patch that fixes a bug?
 
-#### **Do you intend to add a new feature or change an existing one?**
+* Open a new GitHub pull request which includes your changes.
+* Please include a description which clearly describes the change. Include the relevant issue number if applicable.
+* Before submitting, you can test your code using [pre-commit](https://pre-commit.com/) to validate your pull request will pass our set of tests.
```
Comment (fzipi) [inline CONTRIBUTING.md:37]: This text is the same as the one for reporting an issue. Maybe we need to move that one here?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,37 +1,82 @@
...
+## Testing
+
+Coraza uses Go's built-in test tool. Examples (run from the repository root):
+
+- `go test -v`
+- `go test -v -race ` use to enable the built-in data race detector
+- `go test -run TestDefaultWriters -v ./loggers` run all tests loggers package with name substring `TestDefaultWriters`
 
-#### **Do you have questions about the source code?**
+When a pull request is opened CI will run all tests to verify the change.
```
Comment (fzipi) [inline CONTRIBUTING.md:63]: When a pull request is opened,  automation will run performing validations and tests to verify the change.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,37 +1,82 @@
...
+Before submitting you can commit using [pre-commit](https://pre-commit.com) to
```
Comment (fzipi) [inline CONTRIBUTING.md:64]: This looks repeated.

File: README.md [hunk 7]
```diff
@@ -8,62 +11,83 @@
...
 
-```sh
-go test ./...
-go test -race ./...
-```
+The Coraza Project maintains implementations and plugins for the following servers: 
 
-Using pre-commit 
+* [Caddy Reverse Proxy and Webserver Plugin](https://github.com/corazawaf/coraza-caddy) (stable, needs a maintainer)
+* [HAProxy SPOE Plugin](https://github.com/jptosso/coraza-spoa) (preview)
```
Comment (fzipi) [inline README.md:55]: Are these links good? They point to your personal account.

Comment (jptosso) [reply README.md:55]: Indeed, please fix links

**Review by fzipi — APPROVED** [review]

**Pull request merged**